scRNA-seq/06-overrepresentation_analysis.Rmd

```diff
@@ -0,0 +1,400 @@
+|-------------|-----------|-------------|
+| `GeneRatio` | Number of genes in the top 100 marker genes and the specific gene set (also in `Count` column) | Number of genes in the top 100 marker genes and _any_ gene set |
+| `BgRatio`   | Number of genes in the background set and the specific gene set | Number of genes in the background set and _any_ gene set |
+
+#### Visualizing results
+
+We can use a dot plot to visualize our significant enrichment results.
+
+```{r dotplot, live = TRUE}
+enrichplot::dotplot(go_ora_results)
+```
+
+**What cell type do you think the cells from cluster 1 are?**
+
+We can use an [UpSet plot](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4720993/) to visualize the **overlap** between the gene sets that were returned as significant.
+
+```{r upsetplot, live = TRUE}
+enrichplot::upsetplot(go_ora_results)
+```
+
+It's important to keep in mind that gene sets or pathways aren't independent, either!
+This is particularly true of GO terms, which are hierarchical in nature.
+
+### CellMarker ORA
+
+Next, we'll use another resource, CellMarker, as additional support for our conclusions based on GO.
+
+#### About CellMarker
+
+[CellMarker](http://bio-bigdata.hrbmu.edu.cn/CellMarker/) is a manually curated database of cell marker data for human and mouse ([Zhang *et al.* 2018](https://doi.org/10.1093/nar/gky900)).
+Studies/evidence of the following nature were included in the manual curation process ([Zhang *et al.* 2018](https://doi.org/10.1093/nar/gky900)):
+
+* scRNA-seq
+* Flow cytometry
+* Immunostaining
+
+And there are both gene and protein markers.
+
+Let's read in the CellMarker data we've already cleaned:
+
+```{r read_in_cm, live = TRUE}
+cellmarker_df <- readr::read_tsv(cellmarker_file)
+```
+#### Run `enricher()`
+
+`clusterProfiler::enricher()` is a more general way to perform ORA -- we specify the pathways or gene sets ourselves.
+We can use the it to run ORA with the CellMarker data, which contains gene symbols.
+That's why we needed to perform gene identifier conversion earlier!
+
+```{r cm_ora}
+cellmarker_ora_results <- enricher(
+  gene = cluster01_markers,  # Genes of interest
+  pvalueCutoff = 0.05,  # More permissive cut off
+  pAdjustMethod = "BH",  # FDR
+  universe = background_set,  # Background set
+  # The pathway information should be a data frame with a term name or 
+  # identifier and the gene identifiers
+  TERM2GENE = cellmarker_df
+)
+```
+
+#### Visualizing results
+
+Let's look at an UpSet plot of the significant results.
+
+```{r cm_upsetplot, live = TRUE}
+enrichplot::upsetplot(cellmarker_ora_results)
+```
+
+One thing that the [web version of CellMarker](http://bio-bigdata.hrbmu.edu.cn/CellMarker/) allows us to do is to get a better idea of _marker prevalence_.
+Some of the genes that show up in our top 100 may not be very specific. 
+It's also likely that Kidney is not very relevant to our analysis of Hodgkin's lymphoma, but Lymph node is highly relevant.
```

Comment:
Two other relevant notes: One is just that I expect blood cells to accumulate in the kidney (it is filtering the blood after all!), so the fact that there is a lot of overlap should not be too surprising. 

But even more relevant might be this plot which I happened to stumble on: http://bio-bigdata.hrbmu.edu.cn/CellMarker/browse.jsp There are a lot of marker genes for the kidney in that database. A lot! Emphasizing bias in reference sources seems good.

Reply:
Will add these points to the text!